### ISSUE TITLE
Kanban demo - Event handlers get triggered when dragging left but not right

### ISSUE THREAD

**Issue by bebraw** [yahoo/react-dnd-touch-backend#26]
I and @prashcr saw something strange. He implemented a kanban board based on the touch backend. The kanban consists of lanes and notes within lanes you can drag around. A lane name can be edited by clicking its header. The problem is that if you drag a lane to the left, swap, and complete the action, it actually activates lane name edit (header onClick gets triggered). This is something that doesn't happen if you drag to right, though.

See the [thread](http://survivejs.com/webpack_react/building_kanban/#comment-2609068344) for full discussion. Here's the [repo](https://github.com/prashcr/kanban-app/tree/experimental) if you want to have a closer look.

I can't see anything obvious in the implementation myself. Let me know if you need anything else and we'll get it done. :+1:

**Comment by prashcr** [comment]
Yep, and this only occurs on desktop, mobile (Android Chrome) works as expected (doesn't trigger onClick when dragged)

**Comment by longlho** [comment]
hmm I think that's in [react-dnd-html5-backend](https://github.com/gaearon/react-dnd-html5-backend) instead?

**Comment by bebraw** [comment]
> hmm I think that's in react-dnd-html5-backend instead?

It goes through the touch backend. [Source](https://github.com/prashcr/kanban-app/blob/experimental/app/components/App.jsx).

@prashcr That first video is set to private.

**Comment by prashcr** [comment]
Make sure you're on the `experimental` branch of the repo, which is what's linked here.

html5-backend version lives on `master`, and is live at: http://prashcr.xyz/kanban-app

@bebraw My bad, fixed.

**Comment by longlho** [comment]
@bebraw @prashcr 

tl;dr: I'll prob pull mouse support from this, since its behavior is wildly different from touch propagation, in the sense that `mouseup` doesn't stop `click` propagation since those are 2 complete separate events (there's no `tap` event, and `touchend` does stop mouse event simulation properly).

So the gist of it is actually how React handles reordering of children (cc @gaearon). Dragging left to right kills the source element, thus `click` doesn't get triggered (since `mousedown` & `mouseup` doesn't happen in the same element). Doing it the other way, the source element does get removed, which is kinda weird. I'll spend some time digging into React [DOMChildrenOperations](https://github.com/facebook/react/blob/master/src/renderers/dom/client/utils/DOMChildrenOperations.js) itself, but that's my investigation right now.

![gif](https://www.dropbox.com/s/fwseecyks0ykpn8/ltr.gif?raw=1)
